mlir/lib/Analysis/Presburger/Matrix.cpp

```diff
@@ -230,6 +230,47 @@ void Matrix<T>::fillRow(unsigned row, const T &value) {
     at(row, col) = value;
 }
 
+// moveColumns is implemented by moving the columns adjacent to the source range
+// [srcPos, srcPos + num) to their final position. When moving right (i.e.
+// dstPos > srcPos), the range of the adjacent columns is [srcPos + num,
+// dstPos + num). When moving left (i.e. dstPos < srcPos) the range of the
+// adjacent columns is [dstPos, srcPos). First, zeroed out columns are inserted
+// in the final positions of the adjacent columns. Then, the zeroed out columns
+// are swapped with the adjacent columns. Finally, the now redundant adjacent
+// columns are deleted.
```

Comment:
Would suggest the following rephrasings:
l. 234: // to their final position. When moving right (i.e.
l. 239: // are copied to their final positions, filling in the zeroes. Finally, the now redundant adjacent

Reply:
@Abhinav271828 not sure I understand
> l. 234: // to their final position. When moving right (i.e.

Reply:
(it's the same as what I wrote)

Reply:
ah, I meant delete the `[srcPos, srcPos + num)`. Because when you say "the columns adjacent to the source range [srcPos, srcPos + num)", imo, it's not clear whether `[srcPos, srcPos + num)` refers to "the source range" or "the columns adjacent to the source range"

Reply:
@Abhinav271828 re:
> l. 239: // are copied to their final positions, filling in the zeroes
I feel it is technically inaccurate to say copy (since we are swapping). How does the following sound instead:
> Then, the adjacent columns are moved to their final positions by swapping them with the zeroed columns.